### ISSUE TITLE
Should we panic when decoding compact size from within a `Witness`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3281]
In #3259 it was observed that we convert an error to an option when decoding a compact size. Should we panic assuming that  the `Witness` is valid.

- Our `serde` deserialization implementation does not check validity (#1327)

ref: https://github.com/rust-bitcoin/rust-bitcoin/pull/3259#discussion_r1737827141

**Comment by apoelstra** [comment]
Yes, we should assume that `Witness`, once constructed, has internally valid data.

I guess we need to fix the serde impl.

**Comment by Kixunil** [comment]
There's no validation required in `serde` impl because the format is equivalent to `Vec<Vec<u8>>` and we simply *compute* the correct values for compact sizes.

**Comment by tcharding** [comment]
I thought I responded to this, had a fun and educational dive into how we serialize / consenus encode the `Witness` resulting in #3286.

You are totally correct @Kixunil, #3259 now just panics if we cannot parse a compact size encoding ie., `Witness` is invalid.
